### PR TITLE
[REVIEW] Implement drop_duplicates for Series

### LINKED ISSUE #2233: [FEA] drop_duplicates for Series
**Is your feature request related to a problem? Please describe.**
We seem to have `DataFrame.drop_duplicates` but not `Series.drop_duplicates`. It would be nice to have `drop_duplicates` for series too because `dask.dataframe.categorize()` fails due to lack of this api on Series.

**Describe the solution you'd like**
```python
import cudf
import pandas as pd
import dask
import dask_cudf
df = pd.DataFrame({ "C":[1,2,3,4,5]})
ddf = dask.dataframe.from_pandas(df, npartitions=10)
print(ddf.C)
Dask Series Structure:
npartitions=4
0    int64
1      ...
2      ...
3      ...
4      ...
Name: C, dtype: int64
Dask Name: getitem, 8 tasks

print(ddf.categorize(columns=['C']).C)
Dask Series Structure:
npartitions=4
0    category[known]
1                ...
2                ...
3                ...
4                ...
Name: C, dtype: category
Dask Name: getitem, 32 tasks

df = cudf.from_pandas(df)
ddf1 = dask_cudf.from_cudf(df, npartitions=10)
print(ddf1.C.dtype)
int64

print(ddf1.categorize(columns=['C']).C)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/dask/dask/dataframe/core.py", line 3497, in categorize
    self, columns=columns, index=index, split_every=split_every, **kwargs
  File "/dask/dask/dataframe/categorical.py", line 140, in categorize
    categories, index = compute_as_if_collection(type(df), dsk, (prefix, 0), **kwargs)
  File "/dask/dask/base.py", line 202, in compute_as_if_collection
    return schedule(dsk2, keys, **kwargs)
  File "/dask/dask/local.py", line 532, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "/dask/dask/local.py", line 476, in get_async
    fire_task()
  File "/dask/dask/local.py", line 471, in fire_task
    callback=queue.put,
  File "/dask/dask/local.py", line 521, in apply_sync
    res = func(*args, **kwds)
  File "/dask/dask/local.py", line 238, in execute_task
    result = pack_exception(e, dumps)
  File "/dask/dask/local.py", line 233, in execute_task
    result = _execute_task(task, data)
  File "/dask/dask/core.py", line 119, in _execute_task
    return func(*args2)
  File "/dask/dask/dataframe/categorical.py", line 47, in _get_categories
    res[col] = x.dropna().drop_duplicates()
AttributeError: 'Series' object has no attribute 'drop_duplicates'
```

**Describe alternatives you've considered**
We currently don't support `astype('category')` too, only alternative is to convert to dask or pandas -> convert to category -> convert entire dataframe to cudf/dask-cudf.

**Additional context**
Dask `.categorize()` api fails because of this. If `.categorize()` fails we will still be able to use `get_dummies` but by using the above stated workaround.

### PR DESCRIPTION
**Pull request by galipremsagar**
This resolves #2233

### CODE REVIEW

File: python/cudf/cudf/dataframe/series.py [hunk 1]
```diff
@@ -1069,6 +1072,32 @@ def masked_assign(self, value, mask):
...
+        in_cols = [self._column]
+        in_index = self.index
+        from cudf.dataframe.multiindex import MultiIndex
+        if isinstance(in_index, MultiIndex):
+            in_index = RangeIndex(len(in_index))
+        out_cols, new_index = cpp_drop_duplicates(
+            [in_index.as_column()], in_cols, None, keep
+        )
+        new_index = as_index(new_index)
+        if len(self.index) == len(new_index) and self.index.equals(new_index):
```
Comment (kkraus14) [inline python/cudf/cudf/dataframe/series.py]: I believe this length check is already done in the `self.index.equals` call so there's no need to do it here for optimization.

Comment (kkraus14) [reply python/cudf/cudf/dataframe/series.py]:
```suggestion
        if self.index.equals(new_index):
```

Comment (galipremsagar) [reply python/cudf/cudf/dataframe/series.py]: @kkraus14, done. Removed the length check.

**Comment by kkraus14** [comment]
@galipremsagar the issue is that the length check and reference checks aren't done for `RangeIndex`, can you add them?

**Comment by galipremsagar** [comment]
> @galipremsagar the issue is that the length check and reference checks aren't done for `RangeIndex`, can you add them?

@kkraus14 Added the checks for `RangeIndex`

**Pull request merged**